src/core/qgsproject.cpp

```diff
@@ -975,6 +983,17 @@ void QgsProject::onMapLayersAdded( const QList<QgsMapLayer*>& layers )
     }
 
     connect( layer, SIGNAL( configChanged() ), this, SLOT( setDirty() ) );
+
+    // check if we have to update connections for layers with dependencies
+    for ( QMap<QString, QgsMapLayer*>::iterator it = existingMaps.begin(); it != existingMaps.end(); it++ )
+    {
+      QSet<QString> deps = it.value()->dataDependencies();
+      if ( deps.contains( layer->id() ) )
+      {
+        // reconnect to change signals
+        it.value()->setDataDependencies( deps );
```

Comment:
I am not sure if I understand why this needs to happen. Aren't dependencies dismissed in existing layers if they were not found previously?